src/v/archival/ntp_archiver_service.cc

```diff
@@ -2608,6 +2678,112 @@ ss::future<> ntp_archiver::apply_spillover() {
     }
 }
 
+flush_result ntp_archiver::flush() {
+    // Return early if we are not the leader, or if we are a read replica.
+    if (!_parent.is_leader() || _parent.is_read_replica_mode_enabled()) {
+        vlog(
+          _rtclog.debug,
+          "Flush request was not accepted, as partition is not the leader or "
```

Comment:
> partition is not the leader

nit: node/replica is not leader for partition

Reply:
Corrected.